src/inmanta/agent/in_process_executor.py

```diff
@@ -501,7 +501,8 @@ async def stop(self) -> None:
             await child.stop()
 
     async def start(self) -> None:
-        pass
+        assert all(e.is_stopped() for e in self.executors.values())
```

Comment:
is_stopped doesn't return if the executor is stopped, but if stop was requested.

I wonder if we have to do a join as well? (I don't exactly know why the assert is added)

Reply:
> is_stopped doesn't return if the executor is stopped, but if stop was requested.

Indeed. The naming is rather confusing.

> I wonder if we have to do a join as well? (I don't exactly know why the assert is added)

I just wanted to make sure that if we call `start()` after `stop()` + `join()`, that the executors are actually stopped. So that we can be sure that no executors keep hanging after clearing the executors dictionary. I think this use case is only relevant for the tests.